certbot-dns-godaddy/certbot_dns_godaddy/dns_godaddy.py

```diff
@@ -0,0 +1,168 @@
+        :raises certbot.errors.PluginError: if an error occurs communicating with the Godaddy
+                                            API
+        """
+
+        try:
+            domain = self._find_domain(domain_name)
+        except godaddypy.client.BadResponse as e:
+            hint = None
+
+            if "UNABLE_TO_AUTHENTICATE" in str(e):
+                hint = 'Did you provide a valid API token?'
+
+            logger.debug('Error finding domain using the Godaddy API: %s', e)
+            raise errors.PluginError('Error finding domain using the Godaddy API: {0}{1}'
+                                     .format(e, ' ({0})'.format(hint) if hint else ''))
+
+        try:
+            self.client.add_record(domain, {
+                'data': record_content,
+                'name': self._compute_record_name(domain, record_name),
+                'type': 'TXT'
+            })
+            logger.debug('Successfully added TXT record')
+        except godaddypy.client.BadResponse as e:
+            logger.debug('Error adding TXT record using the Godaddy API: %s', e)
+            raise errors.PluginError('Error adding TXT record using the Godaddy API: {0}'
+                                     .format(e))
+
+    def del_txt_record(self, domain_name, record_name, record_content):
+        """
+        Delete a TXT record using the supplied information.
+
+        Note that both the record's name and content are used to ensure that similar records
+        created concurrently (e.g., due to concurrent invocations of this plugin) are not deleted.
+
+        Failures are logged, but not raised.
+
+        :param str domain_name: The domain to use to associate the record with.
+        :param str record_name: The record name (typically beginning with '_acme-challenge.').
+        :param str record_content: The record content (typically the challenge validation).
+        """
+
+        try:
+            domain = self._find_domain(domain_name)
+        except godaddypy.client.BadResponse as e:
+            logger.debug('Error finding domain using the Godaddy API: %s', e)
+            return
+
+        try:
+            domain_records = self.client.get_records(domain, record_type='TXT')
+
+            matching_records = [record for record in domain_records
+                                if record['type'] == 'TXT'
+                                and record['name'] == self._compute_record_name(domain, record_name)
+                                and record['data'] == record_content]
+        except godaddypy.client.BadResponse as e:
+            logger.debug('Error getting DNS records using the Godaddy API: %s', e)
+            return
+
+        for record in matching_records:
+            try:
+                logger.debug('Removing TXT record with name: %s', record['name'])
+                self.client.delete_records(domain, name=record['name'], record_type='TXT')
```

Comment:
Unfortunately, Godaddy doesn't provide a good API for this. **records** attribute for PUT request is mandatory, and empty list is not accepted.

If replaced record list empty (in our case, saved record list is going to be empty all the time), it simply fails with 400. It doesn't provide a DELETE request either.